public/main/inc/lib/usergroup.lib.php

```diff
@@ -1209,6 +1211,36 @@ public function unsubscribe_courses_from_usergroup($usergroup_id, $delete_items,
         }
     }
 
+    public function unsubscribe_only_courses_from_usergroup($usergroup_id, $delete_items, $sessionId = 0)
+    {
+        $sessionId = (int) $sessionId;
+        // Deleting items.
+        if (!empty($delete_items)) {
+            $user_list = $this->get_users_by_usergroup($usergroup_id);
+
+            $groupId = isset($_GET['id']) ? (int) $_GET['id'] : 0;
+            foreach ($delete_items as $course_id) {
+                $course_info = api_get_course_info_by_id($course_id);
+                if ($course_info) {
+                    Database::delete(
+                        $this->usergroup_rel_course_table,
+                        [
+                            'usergroup_id = ? AND course_id = ?' => [
+                                $usergroup_id,
+                                $course_id,
+                            ],
+                        ]
+                    );
+                }
+                if (0 != $sessionId && 0 != $groupId) {
+                    $this->subscribe_sessions_to_usergroup($groupId, [0]);
```

Comment:
You cannot add a "subscribe*()" call inside a method that is called "unsubscribe_only_courses_from_usergroup()". What is the goal here ?